Error:Gradle project cmake.path is ...\app\build\secure-keys\include\main\cpp\CMakeLists.txt but that file doesn't exist

### Description
I receive the error given in issue title when trying to compile a newly created project

### Reproduction
Just create a new project in Android Studio and follow the steps mentioned in wiki to include the library

### Additional Information

* library version : 2.1.0
* android studio : 3.0 Canary 5
* compileSdkVersion : 26
* buildToolsVersion : 26.0.0
* minSdkVersion : 18
* targetSdkVersion : 26

I tried `gradlew build` but got the same error
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':testapp:generateJsonModelDebug'.
> Gradle project cmake.path is C:\android-api-SecureKeys\testapp\build\secure-keys\include\main\cpp\CMakeLists.txt but that file doesn't exist

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

/Users/gars/Library/Android/sdk/ndk-bundle/sysroot/usr/include/linux/types.h:21:10: fatal error: 'asm/types.h' file not found
  #include <asm/types.h>
           ^~~~~~~~~~~~~
  1 error generated.
  ninja: build stopped: subcommand failed.

Yes, thats the usual issue of this thread, I've opened a PR (which still requires further testing before merging it). As soon as I have time again I will do so and release a new version.

Your older comment was about an issue not finding an inner header of the ndk-bundle
```
In file included from /Users/gars/Library/Android/sdk/ndk-bundle/sysroot/usr/include/wchar.h:32:
In file included from /Users/gars/Library/Android/sdk/ndk-bundle/sysroot/usr/include/stdio.h:42:
In file included from /Users/gars/Library/Android/sdk/ndk-bundle/sysroot/usr/include/sys/types.h:36:
/Users/gars/Library/Android/sdk/ndk-bundle/sysroot/usr/include/linux/types.h:21:10: fatal error: 'asm/types.h' file not found
#include <asm/types.h>
```

--- a/plugin/src/main/groovy/com/saantiaguilera/securekeys/gradle/NativePackagerPlugin.groovy
+++ b/plugin/src/main/groovy/com/saantiaguilera/securekeys/gradle/NativePackagerPlugin.groovy
@@ -7,22 +7,11 @@ import org.gradle.api.tasks.bundling.Zip
 /**
  * Created by saantiaguilera on 6/22/17.
  */
-public class NativePackagerPlugin implements Plugin<Project> {
+class NativePackagerPlugin implements Plugin<Project> {
 
     private static final String AAR = 'aar'
 
-    private static final String TASK_EXTRACT_NATIVE_FILES = 'extractSecureKeysNativeFiles'
     private static final String TASK_PACKAGE_NATIVE_FILES = 'packageNativeFilesToAar'
-
-    private static final List<String> TASK_DEPENDANTS = [
-            'generate',
-            'compile',
-            'assemble',
-            'implementation',
-            'api',
-            'compileOnly',
-            'runtimeOnly'
-    ]
 
     private static final String SECUREKEYS_PACKAGE_NAME = 'com.saantiaguilera.securekeys/core'
     private static final String SECUREKEYS_PACKAGE_LOCAL_NAME = 'SecureKeys/testapp/libs/core'
@@ -49,7 +38,7 @@ public class NativePackagerPlugin implements Plugin<Project> {
             }
         } else {
             // Add tasks for retrieveing native files from aar
-            addNativeAarRetrievalTasks(project)
+            addNativeAarRetrieval(project)
             // Add to the project the flags for building the native library
             addNativeBuildingSupport(project)
         }
@@ -87,7 +76,10 @@ public class NativePackagerPlugin implements Plugin<Project> {
     }
 
     def addNativeBuildingSupport(Project project) {
-        if (!project.properties.dontCompileNdk) {
+        // This is because android studio sync wont trigger any gradle task,
+        // Which means our native aar retrieval wont be run, and this will break it.
+        if (!project.gradle.startParameter.taskNames.toListString().contentEquals("[]") &&
+                !project.properties.dontCompileNdk) {
             project.android.defaultConfig.externalNativeBuild {
                 cmake {
                     cppFlags CPP_FLAGS
@@ -101,36 +93,25 @@ public class NativePackagerPlugin implements Plugin<Project> {
         }
     }
 
-    def addNativeAarRetrievalTasks(Project project) {
-        project.task(TASK_EXTRACT_NATIVE_FILES) {
-            doLast {
-                project.configurations.findAll {
-                    project.gradle.gradleVersion >= '4.0' ?
-                    it.isCanBeResolved() :
-                    true
-                }.each { config ->
-                    config.files.each {
-                        def file = it.absoluteFile
-                        // Check if the file is of the package name. 
-                        // Also we check if its our local environment since we dont add it via gradle
-                        if (file.absolutePath.contains(SECUREKEYS_PACKAGE_NAME) ||
-                                file.absolutePath.contains(SECUREKEYS_PACKAGE_LOCAL_NAME)) {
-                            project.copy {
-                                from project.zipTree(file)
-                                into AAR_UNCOMPRESSED_ROOT_DESTINATION
-                                include "${AAR_GENERATED_FOLDER}/${FILE_BLOB_ALL}"
-                            }
+    def addNativeAarRetrieval(Project project) {
+        project.afterEvaluate {
+            project.configurations.findAll {
+                project.gradle.gradleVersion >= '4.0' ?
+                        it.isCanBeResolved() :
+                        true
+            }.each { config ->
+                config.files.each {
+                    def file = it.absoluteFile
+                    // Check if the file is of the package name.
+                    // Also we check if its our local environment since we dont add it via gradle
+                    if (file.absolutePath.contains(SECUREKEYS_PACKAGE_NAME) ||
+                            file.absolutePath.contains(SECUREKEYS_PACKAGE_LOCAL_NAME)) {
+                        project.copy {
+                            from project.zipTree(file)
+                            into AAR_UNCOMPRESSED_ROOT_DESTINATION
+                            include "${AAR_GENERATED_FOLDER}/${FILE_BLOB_ALL}"
                         }
                     }
-                }
-            }
-        }
-
-        project.tasks.build.dependsOn(TASK_EXTRACT_NATIVE_FILES)
-        project.tasks.whenTaskAdded { task ->
-            TASK_DEPENDANTS.each {
-                if (task.name.toLowerCase().contains(it.toLowerCase())) {
-                    task.dependsOn(TASK_EXTRACT_NATIVE_FILES)
                 }
             }
         }

